app/api/v0_1/demand.py

```diff
@@ -1,8 +1,119 @@
 from . import bp
+from . import errors
 
-from app.models import Demand
+from flask import request, jsonify
+
+from typing import Dict,Union
+
+from app import db
+
+from app.models import Demand, Unit
 
 
 @bp.route('/demand', methods=['POST'])
 def demand():
-    return 'demand placeholder'
\ No newline at end of file
+
+    if not request.is_json:
+        raise errors.InvalidUsage("Incorrect request format! Request data must be JSON")
+
+    data = request.get_json()
+
+    # check_demands(data['demands'])
+
+    if "demands" in data and data["demands"]:
+        demands = data['demands']
+    else:
+        raise errors.InvalidUsage("'demands' missing in request data")
+
+
+    params = ["latitude","longitude", "cluster_id", "unit_name", "quantity"]
+
+    # Checking if each element is valid
+    for demand in demands:
+
+        error = check_demand(demand)
+        if error:
+            return error
+
+        # Filtering the dict for safety
+        demand = {param:demand[param] for param in params}
+
+    demand_entries = []
+
+    # Adding demands to database
+    for demand in demands:
+        unit = Unit.query.filter_by(name=demand['unit_name']).first()
+        demand_entry = Demand(latitude=demand['latitude'],longitude=demand['longitude'],units=demand['quantity'],cluster_id=demand['cluster_id'],unit_id=unit.id)
+        db.session.add(demand_entry)
+        demand_entries.append(demand_entry)
+
+    db.session.commit()
+
+    for demand,demand_entry in zip(demands, demand_entries):
+        demand['id'] = demand_entry.id
+
+    # for unit in data['add_units']:
+    #     print(f"Adding unit '{unit}'")
+    #     un = Unit(name=unit)
+    #     db.session.add(un)
+    #     db.session.commit()
+
+
+    return jsonify(demands)
```

Comment:
more of a question: do we ```return``` ```jsonify(demands)``` for all methods or just GET?

Reply:
Added it to confirm stuff was happening. Its returning the sent objects back with the assigned ID.